packages/inter-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -899,12 +671,165 @@ export const prepareVaultManagerKit = (
         getPublicFacet() {
           return this.facets.collateral;
         },
+
+        lockOraclePrices() {
+          const { state, facets } = this;
+          const { self } = facets;
+
+          // XXX: 'twould be better if collateralQuote were updated lazily when it
+          // changes by more than X%, and we use a cached value.  see #6946
+          return E.when(self.getCollateralQuote(), quote => {
+            trace(`lockPrice`, getAmountIn(quote), getAmountOut(quote));
+            // @ts-expect-error declared PriceQuote | undefined. TS thinks undefined.
+            state.lockedQuote = quote;
+            return quote;
+          });
+        },
+        async liquidateVaults(auctionPF) {
+          const { state, facets } = this;
+          const { lockedQuote, compoundedInterest } = state;
+
+          assert(factoryPowers && prioritizedVaults && zcf);
+          lockedQuote ||
+            Fail`Must have locked a quote before liquidating vaults.`;
+          assert(lockedQuote); // redundant with previous line
```

Comment:
I think this migh twork
```suggestion
          lockedQuote ||
            throw Fail`Must have locked a quote before liquidating vaults.`;
```

Reply:
I get a parse failure. "Expression expected". 

I needed the second line to convince typescript that lockedQuote wasn't null.